dali/python/nvidia/dali/ops.py

```diff
@@ -1121,23 +1148,26 @@ def _preprocess_inputs(inputs, op_name, device, schema=None):
     def is_input(x):
         if isinstance(x, (_DataNode, nvidia.dali.types.ScalarConstant)):
             return True
-        return isinstance(x, (list)) and \
-                any(isinstance(y, _DataNode) for y in x) and \
-                all(isinstance(y, (_DataNode, nvidia.dali.types.ScalarConstant)) for y in x)
+        return (isinstance(x, (list))
+                and any(isinstance(y, _DataNode) for y in x)
+                and all(isinstance(y, (_DataNode, nvidia.dali.types.ScalarConstant)) for y in x))
 
     default_input_device = "gpu" if device == "gpu" else "cpu"
 
     for idx, inp in enumerate(inputs):
         if not is_input(inp):
-            input_device = schema.GetInputDevice(idx) or default_input_device if schema else default_input_device
+            if schema:
+                input_device = schema.GetInputDevice(idx) or default_input_device
+            else:
+                input_device = default_input_device
             if not isinstance(inp, nvidia.dali.types.ScalarConstant):
                 try:
                     inp = _Constant(inp, device=input_device)
                 except Exception as ex:
                     raise TypeError("""when calling operator {0}:
 Input {1} is neither a DALI `DataNode` nor a list of data nodes but `{2}`.
-Attempt to convert it to a constant node failed."""
-                    .format(op_name, idx, type(inp).__name__)) from ex
+Attempt to convert it to a constant node failed.""".format(op_name, idx,
+                                                           type(inp).__name__)) from ex
```

Comment:
I don't think we support anything older than Python 3.6, do we?
```suggestion
                   raise TypeError(f"""when calling operator {op_name}:
Input {idx} is neither a DALI `DataNode` nor a list of data nodes but `{type(inp).__name__)}`.
Attempt to convert it to a constant node failed.""") from ex
```